src/schemaDirectives/directiveResolvers.ts

```diff
@@ -0,0 +1,25 @@
+export const directiveResolvers = {
+  auth: (next: Function) => (root: any, args: any, context: any, info: any) => {
+    if (!context.auth || !context.auth.isAuthenticated()) {
+      throw new Error(`User not Authenticated`)
+    }
+    return next(root, args, context, info)
+  },
+  hasRole: (roles: Array<string>) => (next: Function) => (root: any, args: any, context: any, info: any) => {
+    if (!context.auth || !context.auth.isAuthenticated()) {
+      throw new Error(`User not Authenticated`)
+    }
+
+    let foundRole = null // this will be the role the user was successfully authorized on
+
+    foundRole = roles.find((role: string) => {
```

Comment:
Actually this will be vulnerable to code swaping. Using classical for loop may be more secure